controllers/apiserver.go

```diff
@@ -39,6 +45,82 @@ var samplePipelineTemplates = map[string]string{
 	"sample-config":   "apiserver/sample-pipeline/sample-config.yaml.tmpl",
 }
 
+func (r *DSPAReconciler) GenerateSamplePipelineMetadataBlock(pipeline string) (map[string]string, error) {
+
+	item := make(map[string]string)
+
+	// Get Required Fields
+	pName, err := config.GetStringConfigOrError(fmt.Sprintf("ManagedPipelinesMetadata.%s.Name", pipeline))
+	if err != nil {
+		return nil, err
+	}
+	pFile, err := config.GetStringConfigOrError(fmt.Sprintf("ManagedPipelinesMetadata.%s.Filepath", pipeline))
+	if err != nil {
+		return nil, err
+	}
+	platformVersion := config.GetStringConfigWithDefault("PlatformVersion", config.DefaultPlatformVersion)
+
+	// Get optional fields
+	pDesc := config.GetStringConfigWithDefault(fmt.Sprintf("ManagedPipelinesMetadata.%s.Description", pipeline), "")
+	pVerName := config.GetStringConfigWithDefault(fmt.Sprintf("ManagedPipelinesMetadata.%s.VersionName", pipeline), "")
+	pVerDesc := config.GetStringConfigWithDefault(fmt.Sprintf("ManagedPipelinesMetadata.%s.VersionDescription", pipeline), "")
+
+	// Create Sample Config item
+	item["name"] = pName
+	item["file"] = pFile
+	item["description"] = pDesc
+	item["versionName"] = fmt.Sprintf("%s - %s", pVerName, strings.Trim(platformVersion, "\""))
```

Comment:
Is the trim because the platform is passing the version as a JSON marshaled value or is Viper not parsing it correctly?

Reply:
if the config passed in is `PLATFORMVERSION="v1.2.3"`, the ENV VAR ends up being `"v1.2.3"` and this breaks templating as @HumairAK noted [here](https://github.com/opendatahub-io/data-science-pipelines-operator/pull/781#discussion_r1954673570)